clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -417,6 +471,11 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
         }
 #endif
         shard->prev_instr_ = memref;
+        if (cur_instr_decoded != nullptr) {
```

Comment:
Done.

Reply:
Removing the ` if (cur_instr_decoded != nullptr)` on L433 results in a false `"Direct branch does not go to the correct target",  "Failed to catch branch not going to its target"` error.  Investigating why that's happening.

Reply:
This is because `shard->prev_instr_decoded_` won't be updated with just a check in the inner loop since we always start from a state where `shard->prev_instr_decode_` was a `nullptr` and we are checking if it != nullptr before updating it. 

```
        if (shard->prev_instr_decoded_ != nullptr) {
            instr_free(GLOBAL_DCONTEXT, shard->prev_instr_decoded_.get());
            shard->prev_instr_decoded_ = std::move(cur_instr_decoded);
        }
```

Reply:
Nevermind, I misunderstood the comment above. Just removing `(cur_instr_decoded != nullptr` and moving the update outside of the if-block works. 
```
if (shard->prev_instr_decoded_ != nullptr) 
    instr_free(GLOBAL_DCONTEXT, shard->prev_instr_decoded_.get()); 
shard->prev_instr_decoded_ = std::move(cur_instr_decoded);
```